Libs/DICOM/Widgets/ctkDICOMPatientItemWidget.cpp

```diff
@@ -357,20 +366,249 @@ void ctkDICOMPatientItemWidgetPrivate::createStudies()
+  QAbstractItemModel* model = this->PatientServersCheckableComboBox->checkableModel();
+  int wasBlocking = this->PatientServersCheckableComboBox->blockSignals(true);
+
+  this->PatientServersCheckableComboBox->clear();
+  this->PatientServersCheckableComboBox->addItems(allActiveConnectionNames);
+  this->EnabledServers.clear();
+  for (int filterIndex = 0; filterIndex < this->PatientServersCheckableComboBox->count(); ++filterIndex)
+  {
+    QString connectionName = this->PatientServersCheckableComboBox->itemText(filterIndex);
+    QModelIndex modelIndex = model->index(filterIndex, 0);
+
+    Qt::CheckState checkState = Qt::CheckState::PartiallyChecked;
+    if (allowList.contains(connectionName))
+    {
+      checkState = Qt::CheckState::Checked;
+    }
+    else if (denyList.contains(connectionName))
+    {
+      checkState = Qt::CheckState::Unchecked;
+    }
+
+    this->PatientServersCheckableComboBox->setCheckState(modelIndex, checkState);
+
+    if (checkState != Qt::CheckState::Unchecked)
+    {
+      this->EnabledServers.append(connectionName);
+    }
+  }
+  this->PatientServersCheckableComboBox->blockSignals(wasBlocking);
+
+  this->saveEnabledServersStringListToChildren(this->EnabledServers);
+}
+
+//----------------------------------------------------------------------------
+bool ctkDICOMPatientItemWidgetPrivate::areAllActiveServersSecure()
+{
+  QAbstractItemModel* model = this->PatientServersCheckableComboBox->checkableModel();
+  for (int filterIndex = 0; filterIndex < this->PatientServersCheckableComboBox->count(); ++filterIndex)
+  {
+    QModelIndex modelIndex = model->index(filterIndex, 0);
+    Qt::CheckState checkState = this->PatientServersCheckableComboBox->checkState(modelIndex);
+    if (checkState == Qt::CheckState::PartiallyChecked)
+    {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+//----------------------------------------------------------------------------
+void ctkDICOMPatientItemWidgetPrivate::setAllInsecureServerEnabledStatus(bool enabled)
+{
+  bool wasModifying = this->PatientServersCheckableComboBox->blockSignals(true);
+  QAbstractItemModel* model = this->PatientServersCheckableComboBox->checkableModel();
+  for (int filterIndex = 0; filterIndex < this->PatientServersCheckableComboBox->count(); ++filterIndex)
+  {
+    QModelIndex modelIndex = model->index(filterIndex, 0);
+    Qt::CheckState checkState = this->PatientServersCheckableComboBox->checkState(modelIndex);
+    if (checkState != Qt::CheckState::PartiallyChecked)
+    {
+      continue;
+    }
+
+    checkState = enabled ? Qt::CheckState::Checked : Qt::CheckState::Unchecked;
+    this->PatientServersCheckableComboBox->setCheckState(modelIndex, checkState);
+  }
+  this->PatientServersCheckableComboBox->blockSignals(wasModifying);
+  this->saveEnabledServersStringListFromUI();
+}
+
+//----------------------------------------------------------------------------
+bool ctkDICOMPatientItemWidgetPrivate::askUserActionForServerSecurity()
```

Comment:
not security - remove this from everywhere else (in the future maybe we'll introduce the concept of secure or insecure servers, but allow/deny is not that)